chain/network/src/peer_manager/tests/routing.rs

```diff
@@ -838,6 +838,11 @@ async fn max_num_peers_limit() {
 
     tracing::info!(target:"test", "wait for {id3} routing table");
     pm3.wait_for_routing_table(&[]).await;
+
+    drop(pm0);
```

Comment:
comment why do you have to drop them explicitly now.

Reply:
This is a nasty subject. AFAIU NLL (https://blog.rust-lang.org/2022/08/05/nll-by-default.html ) this is NOT strictly necessary, as the destructors of these values should be called exactly at the end of the function anyway (unless the problem is even deeper: for example the compiler incorrectly figures out that it can reorder some instructions). However, I add those every time I observe some flakiness that I can't reproduce locally. 

Our CI setup sucks, because:
1. it runs cargo test, which doesn't support "[panic=abort]"
2. I suspect that these tests have some thread panicking, but that doesn't stop test execution because of (1)
3. As a result the whole presubmit times out (because cargo test doesn't even support setting timeouts on tests) and we get no logs.

I add these explicit drops to just eliminate a possibility that I just don't understand how NLL works.

Reply:
Thanks for detailed info. Please put some of it in the comment in the code (so that the next reader can understand why you did it).

Reply:
done